projects/admin/src/app/circulation/items.service.ts

```diff
@@ -66,21 +66,19 @@ export class ItemsService {
     }
     return this.http.get<any>(url).pipe(
       map(data => {
-        return data.metadata;
-      }),
-      map(data => {
-        const item = new Item(data.item);
-        if (data.loan) {
-          item.setLoan(data.loan);
+        const item = new Item(data.metadata.item);
+        if (data.metadata.loan) {
+          item.setLoan(data.metadata.loan);
         }
         return item;
       }),
       catchError(e => {
         if (e.status === 404) {
-          return of(null);
+          return Promise.reject(new Error('item not found!'));
+          // return of(null);
         }
```

Comment:
What happens if error status is different than 404 ? I think the other cases have to be handled.

Reply:
done